airbyte-integrations/connectors/source-datadog/source_datadog/streams.py

```diff
@@ -0,0 +1,288 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Iterable, List, Mapping, MutableMapping, Optional, Union
+
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams import IncrementalMixin
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class DatadogStream(HttpStream, ABC):
+    """
+    Datadog API Reference: https://docs.datadoghq.com/api/latest/
+    """
+
+    primary_key: Optional[str] = None
+    parse_response_root: Optional[str] = None
+
+    def __init__(self, query: str, limit: int, start_date: str, end_date: str, **kwargs):
+        super().__init__(**kwargs)
+        self.query = query
+        self.limit = limit
+        self.start_date = start_date
+        self.end_date = end_date
+        self._cursor_value = None
+
+    @property
+    def url_base(self) -> str:
+        return "https://api.datadoghq.com/api"
+
+    def request_headers(self, **kwargs) -> Mapping[str, Any]:
+        return {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        }
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params: Dict[str, str] = {}
+
+        if next_page_token:
+            params.update(next_page_token)
+
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        records = response_json if not self.parse_response_root else response_json.get(self.parse_response_root, [])
+        for record in records:
+            yield self.transform(record=record, **kwargs)
+
+    def transform(self, record: MutableMapping[str, Any], stream_slice: Mapping[str, Any], **kwargs) -> MutableMapping[str, Any]:
+        return record
```

Comment:
and for this 2 add unit test too if possible :)

Reply:
Done